services/metadata_service/api/artifact.py

```diff
@@ -314,12 +322,16 @@ async def get_artifacts_by_step(self, request):
         artifacts = await self._async_table.get_artifact_in_steps(
             flow_name, run_number, step_name
         )
-
-        filtered_body = filter_artifacts_for_latest_attempt(
-            artifacts.body)
-        return web.Response(
-            status=artifacts.response_code, body=json.dumps(filtered_body)
-        )
+        if artifacts.response_code == 200:
+          filtered_body = filter_artifacts_for_latest_attempt(
+              artifacts.body)
+          return web.Response(
+              status=artifacts.response_code, body=json.dumps(filtered_body)
+          )
+        else:
+          return web.Response(
+            status=500,body=json.dumps(http_500(artifacts.body))
```

Comment:
should it pass through `artifacts.response_code` here?

Reply:
Done.